manifests/charts/base/templates/endpoints.yaml

```diff
@@ -8,7 +8,7 @@ metadata:
   {{- else }}
   name: istiod
   {{- end }}
-  namespace: {{ .Release.Namespace }}
+  namespace: {{ .Values.global.istioNamespace | default .Release.Namespace }}
```

Comment:
A bit confused on how this works with helm - I thought now helm doesn't cross namespace boundary (or is considered a bad practice ) ?

Normally to install istiod in foo namespace you just deploy it there with --namespace foo. The use case is to have each istiod install in a separate ns for a different tenant, etc - we support this already.

Reply:
The goal here is to enable the Helm release (i.e. the secret with the Chart information) to be installed in a different namespace than the istio control plane itself. Because Istio's helm charts use `.Release.Namespace` as an alias for the system, this is not currently possible and breaks scenarios where users may want to centralize all helm releases in a different namespace than the actual compute. Note that cert-manager and prometheus (2 popular cloud native charts) enable this scenario:

Cert manager: `{{.Values.namespace|default.Release.Namespace}}`
Prom: ` .Values.forceNamespace | default .Release.Namespace`